### ISSUE TITLE
Youtube not working

### ISSUE THREAD

**Issue by alfonsomoratalla** [ytdl-org/youtube-dl#32897]
## Checklist

- [x] I'm reporting a broken site support issue
- [X] I've verified that I'm running youtube-dl version **2024.08.06.1919**
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [X] I've searched the bugtracker for similar bug reports including closed ones
- [X] I've read bugs section in FAQ

## Verbose log

```youtube-dl "https://www.youtube.com/watch?v=XY1SU1IH-q0" --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://www.youtube.com/watch?v=XY1SU1IH-q0', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2024.08.06.1919 (single file build)
** This build is unofficial daily builds, provided for ease of use.
** Please do not ask for any support.
[debug] Python 3.10.12 (CPython x86_64 64bit) - Linux-5.15.0-117-generic-x86_64-with-glibc2.35 - OpenSSL 3.0.2 15 Mar 2022 - glibc 2.35
[debug] exe versions: ffmpeg 4.4.2, ffprobe 4.4.2
[debug] Proxy map: {}
[youtube] XY1SU1IH-q0: Downloading webpage
[youtube] XY1SU1IH-q0: Downloading player bd3293c9
WARNING: [youtube] Unable to decode n-parameter: download likely to be throttled (Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output. Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1672, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1644, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/youtube.py", line 1620, in _extract_n_function_name
    nfunc_and_idx = self._search_regex(
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 1013, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
)
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://rr8---sn-vg5obxn25po-8ajs.googlevideo.com/videoplayback?expire=1722960274&ei=MvWxZsifMPeAp-oPo9G_sAY&ip=85.59.216.74&id=o-AP3EKmRPKp8LYsXrqb6lOXtUuN1x39cD9gx2dqE55Q5u&itag=136&aitags=134%2C136%2C160&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=pI&mm=31%2C29&mn=sn-vg5obxn25po-8ajs%2Csn-h5q7knes&ms=au%2Crdu&mv=m&mvi=8&pl=20&gcr=es&initcwndbps=1336250&bui=AXc671KswDlVZS-flSEMfpWCf0fskwdcfQjK01e1yY84LJGFa8VHko7g_9aDslsBXm2F6ogZlBS4vd6p&spc=Mv1m9mvrTWphuH-6LbGKBbpNUDmErWTPUsZT9PUUHzDazf8WsEYz6Kr-lt1a&vprv=1&svpuc=1&mime=video%2Fmp4&ns=bm9e43RutSlvLWPbleIbW0UQ&rqh=1&gir=yes&clen=654637429&dur=5786.360&lmt=1722304333340547&mt=1722938213&fvip=3&keepalive=yes&c=WEB&sefc=1&txp=6209224&n=GcKXyZmbJncg5gFnByy&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cxpc%2Cgcr%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIhAJXms_aHETNNyyVqHYI3XFbadEDfj1BkBzl_5NpwgnhCAiAyJq10uVpaPQtihxIJkQbZ5KXrAReJNDMfr23KktFelA%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AGtxev0wRAIgWpaOtHJsiti5K6xMO0q_NZ-GqmioO9gJsWdwKqmJrNQCIBpdXvQjq2oC2j30zP9SgUTzn3jITMIArRBq-eHHy5wE'
[dashsegments] Total fragments: 63
[download] Destination: Starfighter - La aventura comienza (1984) castellano--Ciencia ficción.-- Acción. --Aventuras ----XY1SU1IH-q0.f136.mp4
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 1 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 2 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 3 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 4 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 5 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 6 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 7 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 8 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 9 of 10)...
[download] Got server HTTP error: HTTP Error 403: Forbidden. Retrying fragment 1 (attempt 10 of 10)...
ERROR: giving up after 10 fragment retries
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/bin/youtube-dl/__main__.py", line 19, in <module>
    youtube_dl.main()
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 475, in main
    _real_main(argv)
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 465, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2088, in download
    res = self.extract_info(
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 818, in extract_info
    return self.__extract_info(url, ie, download, extra_info, process)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 825, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 857, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 891, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1701, in process_video_result
    self.process_info(new_info)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1987, in process_info
    partial_success = dl(fname, new_info)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1932, in dl
    return fd.download(name, info)
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/common.py", line 376, in download
    return self.real_download(filename, info_dict)
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/dash.py", line 73, in real_download
    self.report_error('giving up after %s fragment retries' % count)
  File "/usr/local/bin/youtube-dl/youtube_dl/downloader/common.py", line 175, in report_error
    self.ydl.report_error(*args, **kargs)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 638, in report_error
    self.trouble(error_message, tb)
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 600, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())
```

## Description

Cannot download youtube videos

**Comment by GregoriusT** [comment]
I would like to add that now even Format 18 will cause error 403 for some videos, starting to make youtube-dl unusable again, thanks to youtubes shenanigans...

**Comment by dirkf** [comment]
You are using the [obsolete ytdl-patched build](https://github.com/ytdl-patched/youtube-dl/pull/1); use the nightly build instead, per #30839.

Having said that, the nightly and master code hasn't yet been fixed for #32896.
